ethers-middleware/src/proxy_wallet/mod.rs

```diff
@@ -0,0 +1,23 @@
+mod ds_proxy;
+mod gnosis_safe;
+mod middleware;
+
+use ethers_contract::AbiError;
+use ethers_core::{abi::ParseError, types::*};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ProxyWalletError {
+    #[error("dummy error")]
+    Dummy,
+
+    #[error(transparent)]
+    AbiParseError(#[from] ParseError),
+
+    #[error(transparent)]
+    AbiError(#[from] AbiError),
+}
+
+pub trait ProxyWallet: Send + Sync + std::fmt::Debug {
+    fn get_proxy_tx(&self, tx: TransactionRequest) -> Result<TransactionRequest, ProxyWalletError>;
+}
```

Comment:
I wonder if there's some better naming here, maybe `TransformerMiddleware` and `fn transform`?